modules/CloudFront_S3Website/fragments/template.yaml

```
@@ -0,0 +1,122 @@
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      AccessControl: Private
+      WebsiteConfiguration:
+        IndexDocument: index.html
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+  Oac:
+    Type: AWS::CloudFront::OriginAccessControl
+    Properties: 
+      OriginAccessControlConfig: 
+        Description: !Sub "OAC for ${AWS::StackName}"
+        Name: !Ref AWS::StackName
+        OriginAccessControlOriginType: s3
+        SigningBehavior: always
+        SigningProtocol: sigv4
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref Bucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service: cloudfront.amazonaws.com
+          Action: s3:GetObject
+          Resource: !Sub "${Bucket.Arn}/*"
+          Condition: 
+            StringEquals:
+              AWS:SourceArn: !Sub "arn:${AWS::Partition}:cloudfront::${AWS::AccountId}:distribution/${Distribution}"
+        - Action: "s3:*"
+          Effect: Deny
+          Resource:
+          - !GetAtt Bucket.Arn
+          - !Sub "${Bucket.Arn}/*"
+          Condition:
+            Bool:
+              aws:SecureTranspor: false
+          Principal: "*"
+  Distribution:
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Comment: CDN for S3-backed website
+        Aliases:
+          - !Ref Alias
+        Enabled: "true"
+        DefaultRootObject: index.html
+        HttpVersion: "http2"
+        IPV6Enabled: true
+        CustomErrorResponses:
+          - ErrorCode: 404
+            ResponseCode: 200
+            ResponsePagePath: "/"
+          - ErrorCode: 403
+            ResponseCode: 200
+            ResponsePagePath: "/"
+        DefaultCacheBehavior:
+          AllowedMethods:
+            - GET
+            - HEAD
+          CachedMethods:
+            - GET
+            - HEAD
+          Compress: true
+          ForwardedValues:
+            Cookies:
+              Forward: none
+            QueryString: false
+          TargetOriginId: origin1
+          ViewerProtocolPolicy: redirect-to-https
+        Origins:
+          - ConnectionAttempts: 3
+            ConnectionTimeout: 10
+            DomainName: !GetAtt Bucket.RegionalDomainName
+            Id: "origin1"
+            S3OriginConfig:
+              OriginAccessIdentity: ""
+            OriginAccessControlId: !Ref Oac
+        ViewerCertificate:
+          AcmCertificateArn: !Ref Certificate
+          SslSupportMethod: "sni-only"
+  Dns:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      HostedZoneId: !Ref HostedZoneId
+      Comment: CNAME redirect custom name to CloudFront distribution
+      Name: !Ref Alias
+      Type: CNAME
+      TTL: 900
+      ResourceRecords:
+        - !GetAtt Distribution.DomainName
```

This should allow IPV6 and use an ALIAS record (free requests!) instead of a CNAME
```suggestion
Dns:
    Type: AWS::Route53::RecordSetGroup
    Properties:
      HostedZoneId: !Ref 'HostedZoneId'
      RecordSets:
        - Type: A
          Name: !Ref Alias
          AliasTarget:
            DNSName: !GetAtt 'Distribution.DomainName'
            HostedZoneId: Z2FDTNDATAQYW2
        - Type: AAAA
          Name: !Ref Alias
          AliasTarget:
            DNSName: !GetAtt 'Distribution.DomainName'
            HostedZoneId: Z2FDTNDATAQYW2
```